### ISSUE TITLE
Publish on F-Droid

### ISSUE THREAD

**Issue by mirfatif** [mirfatif/PermissionManagerX#1]
@IzzySoft I want to put this app on F-Droid. Can you help me with license and other requirements (if any)? I'm not very good at understanding licensing issues.

Thanks.

**Comment by IzzySoft** [comment]
I'll try my best.

* **License:** for F-Droid, this must be an OSI approved license. Which one, depends on what you want: rather permissive (extreme: WTFL = "do what the f*** you want with it"), or rather copy-left (free to use for everyone – but always with providing the source, even if you're using it just to provide a service: AGPLv3). Hard to advice without knowing your needs, so let me give you some helpful references to make your choice:
    * Wikipedia: [Comparison of free and open-source software licences](https://en.wikipedia.org/wiki/Comparison_of_free_and_open-source_software_licences) (make sure the "OSI" column in the second table is green for your choice, and ideally the FSF one too)
    * [SPDX License List](https://spdx.org/licenses/): this is what F-Droid links to. Rather meant for picking the right abbreviation and making sure OSI/FSF approval is given
    * [Choose an open source license](https://choosealicense.com/) gives some very short suggestions. Note it doesn't go very deep (and seems to cover a sub-selection of available licenses only – e.g. where it suggests GPLv3, I'd rather suggest either GPLv2 or AGPLv3 as otherwise the tech giants can abuse it in-house in their SaaS segments), but might help you with a pre-selection and to get an idea. Following their link to [More licenses are available](https://choosealicense.com/licenses/) is very helpful again, as it sums up the pros and cons of some licenses nicely – and see, there's the AGPL on top as I'd have suggested :smile:.
    * GNU.org recommendations: [How to choose a license for your own work](https://www.gnu.org/licenses/license-recommendations.en.html) (TL;DR: [here is their list of licenses](https://www.gnu.org/licenses/license-list.html) with short descriptions of the corresponding basics)
    * TL;DR: with "permissive" I'd focus on "public domain"/Unlicense, Apache-2.0 and maybe MIT; otherwise GPLv2 or AGPLv3. But that's just a very raw statement, and I'm no license expert (IANAL etc)…
* other requirements
    * `build.gradle`. For F-Droid, this would need literal values for `versionName` and `versionCode`, as the updater just parses it via RegEx to decide whether to pick up a tag for a new build. You are [currently using placholders](https://github.com/mirfatif/PermissionManagerX/blob/master/app/build.gradle#L14) – neither variables nor function calls will work for F-Droid.
    * Metadata. Consider setting up Fastlane structures in your app's repo so you can maintain summary/description yourself (not mandatory, but very strongly suggested; else you'd need to open MRs for each change here), and also can provide screenshots etc. Be welcome to use my [Fastlane Cheat-Sheet](https://gitlab.com/snippets/1895688), which gets you started much faster than finding, reading and especially understanding the "official docs". The absolute minimum (which I call "starter kit") are `en-US/short_description` and `en-US/full_description` (`en-US` is needed as that's used as fallback for elements missing in other locales; for all other languages, the 2-letter-code suffices – eg. `de` for German instead of `de-DE`, `de-AT` etc).
    * Tags. When you got this far (license chosen, gradle fixed, fastlane available), please start [tagging your releases](https://github.com/mirfatif/PermissionManagerX/releases). A tag tells F-Droid which commit you consider a release, so it will be built.

Before applying to F-Droid, I can offer you to test-drive it [via my repo](https://apt.izzysoft.de/fdroid) – especially if you think it's not yet fully ready for the "common public". Later, you can even decide to ship pre-releases via my repo while the stable ones go via F-Droid.org if you wish (several apps have it that way), have the app stay in my repo the same way (as updates reach my repo much faster) – or, and this is the default, have me remove it from my repo once the app got listed and settled at F-Droid.org.

OK, guess that will keep you busy for a little while :laughing: Feel free to ask if something was unclear, or you need further details/help. Best luck!

**Comment by mirfatif** [comment]
@IzzySoft thanks a lot for your detailed and very helpful response.

* For the license part, though I'm not caring much at the moment about protecting my intellectual work, I'd not want to go with fully permissive model. Copy-left would be better in my opinion. And after going through the documentation, the links you provided and what I learned in past, I think GPLv3 would be fine. But you have preferred AGPLv3 over GPLv3. Can you explain it a bit more?
The other libraries I've used in my app use Apache-2.0 and MIT license. Are there any compatibility issues?
* I've hard-coded `versionCode` and `versionName` in `build.gradle`.
But I call shell scripts from `build.gradle` to build native binaries (from C++ code) and a `.dex` file (from Java code). Built files are saved to standard `assets` directory, then packed with `.apk`. Would that work with F-Droid build system?
* Fastlane metadata structure is all set.
* Git tags created.
* As far as testing is concerned, I've been using the app for a year or so and it's quite stable now (at least for me :). It's also up on Play Store now (may be it's too early).
However app's functionality depends on a number of AOSP hidden APIs and a background process which is run in a way Android doesn't like. So I don't think it'd be ever possible to claim that the app works on all stock, custom and OEM ROMs flawlessly. Also I'm not a that qualified developer to make such claims :smile:
But it'd be really nice if you put the app on your repo and/or app listing.

Thanks.

**Comment by IzzySoft** [comment]
> But you have preferred AGPLv3 over GPLv3. Can you explain it a bit more?

GPLv3 allows anyone using your code to build their own service on it, and then offering that service for money (or even data collection, whatever) without giving back their changes or providing the source of it. AGPL closes that gap/loop-hole: they can still offer their service, but have to provide their variant of code using AGPL again. Google hates that :laughing:

> Would that work with F-Droid build system?

I'm the wrong person to ask that (I'm no packager, I cannot tell). But I'd assume it would.

> But it'd be really nice if you put the app on your repo and/or app listing.

Is it not? Then there wasn't an APK available yet I guess. Will look for it and add it ASAP then.

Btw, unrelated note: After having introduced another "broken" library entry, I've now added a pre-commit hook. Starting with the next library commit, "broken entries" should be a thing of the past – be it due to typos or otherwise.

**Comment by IzzySoft** [comment]
OK, your app should [show up here](https://apt.izzysoft.de/fdroid/index/apk/com.mirfatif.permissionmanagerx) in about 4 hours. I had to adjust your `full_description` a bit for the formatting to fit (for F-Droid it might work as-is, though, as they simply apply `nl2br`), and would add an introductionary sentence before the feature list. If you want, I can do some "polish" and make it a PR here (you'd then also get a `de/short_description.txt` along), just let me know.

Basically, I'd move the first 3 paragraphs *before* the feature list, introducing it with

> *Permission Manager X* is a small app to manage permissions and AppOps for the apps installed on your device. It evolved from a shell script …

I'd also rephrase (and re-format) the second paragraph a little. Took me 3 rounds to grasp it, that can be avoided :smile:

> **Manifest permissions** are normally simply called „permissions“ (e.g. Storage, Camera). **AppOps** (app operations) is a robust framework Android uses as back-end …

How does that sound to you?

**Comment by mirfatif** [comment]
> Took me 3 rounds to grasp it, that can be avoided 😄

English isn't my native language 😞 

> How does that sound to you?

Fantastic. So nice of you.

**Comment by IzzySoft** [comment]
> English isn't my native language

Neither it's mine. But maybe two wrongs make one right? :laughing:

> Fantastic.

OK, so there you go: #2. In about 2.5 hours your app goes live in my repo, so you can see how this would look like (I've backported my description; in my repo, it's full HTML – on updates from here, I'd convert it using a Markdown renderer).

**Comment by mirfatif** [comment]
> Neither it's mine. But maybe two wrongs make one right? 😆

Hopefully.

> PS: In my repo, I've temporarily set the license to AGPL-3.0, assuming that will be what you'll decide for. Should I've been wrong, simply yell and I'll fix it up 🤪

All is well 👍

**Comment by IzzySoft** [comment]
Wonderful, thanks!

**Comment by MuntashirAkon** [comment]
It looks like your project cannot be build using `./gradlew assembleDebug` as you're using custom bash scripts with environment variables that aren't set in the `build.gradle` file of the `:app` module. Therefore, it won't be possible for F-Droid to build it. I've tried setting them manually but it still fails.

**Comment by mirfatif** [comment]
@MuntashirAkon I've added How to Build section in ReadMe.

**Comment by MuntashirAkon** [comment]
> @MuntashirAkon I've added How to Build section in ReadMe.

Thanks. But F-Droid cannot build it because it requires a lot of manual work such as replacing `android.jar` (I'm not sure whether F-Droid's going to allow that), JDK 11 requirements, etc.

**Comment by mirfatif** [comment]
> Thanks. But F-Droid cannot build it because it requires a lot of manual work such as replacing `android.jar` (I'm not sure whether F-Droid's going to allow that), JDK 11 requirements, etc.

Should I consider this statement your opinion or F-Droid's reponse? Are you representing F-Droid, officially or unofficially?

**Comment by IzzySoft** [comment]
@mirfatif consider it a "hint". I'm no dev or packager either, but JARs in the source repo usually are show-stoppers. Muntashir is a very active app developer jumping in here to help (which we both certainly appreciate).

If that specific JAR indeed is a show-stopper, we'll find out when either the issue-bot or a packager complain :wink: But the only JAR I know being safe to ignore is that of the gradle-wrapper, as our build process drops that anyway (and replaces it with ours).

**Comment by mirfatif** [comment]
@IzzySoft @MuntashirAkon I'm thankful to both of you for your time.

Replacing `android.jar` is required due to the limitation of the official Android SDK i.e. it doesn't provide access to Android's hidden/internal APIs (in their opinion common app developers should not use them). So the compiler throws `cannot find symbol` errors with official SDK.

Modified `android.jar` is a compile time only dependency. It provides no additional APIs on its own. All APIs are part of Android framework and available at runtime (in `/system/framework/framework.jar` and possibly other `.jar` files).

If there exists some better F-Droid's way to access Android's hidden APIs (or to handle compile time errors ascribed to missing APIs), I can go for that too.

I think someone from F-Droid directly dealing with compiling issues can comment better on this.

**Comment by MuntashirAkon** [comment]
> Replacing `android.jar` is required due to the limitation of the official Android SDK i.e. it doesn't provide access to Android's hidden/internal APIs (in their opinion common app developers should not use them). So the compiler throws `cannot find symbol` errors with official SDK.

That's a valid point actually. You can include required hidden APIs directly in your project but it will be somewhat limited.

> I think someone from F-Droid directly dealing with compiling issues can comment better on this.

I agree. You should open an RFP here: https://gitlab.com/fdroid/rfp or open an MR here: https://gitlab.com/fdroid/fdroiddata if not already.

**Comment by IzzySoft** [comment]
Thanks for the details, Irfan! Would it be possible to provide the sources required to build that JAR? That could either be part of your repo (as Muntashir just suggested) – but it could also be "linked" into your app's repo as git submodule, made into an F-Droid `srclib`, or (if available) pulled from one of the [trusted maven repos](https://gitlab.com/fdroid/fdroidserver/blob/master/fdroidserver/scanner.py#L137).

**Comment by mirfatif** [comment]
> Would it be possible to provide the sources required to build that JAR?

Yes. The whole AOSP code.

**Comment by IzzySoft** [comment]
Ouch. If that means we'd have to check out that just to build this JAR, I guess that won't happen anytime soon. That then only leaves a trusted maven repo.

**Comment by MuntashirAkon** [comment]
> That then only leaves a trusted maven repo.

The `android.jar` used here is available on jitpack.io, I believe. But that's not the problem (it's not the entire AOSP either, just the base (frameworks) repo which can be built explicitly). The problem is this `android.jar` has to be replaced within the SDK rather than the project itself.

**Comment by IzzySoft** [comment]
That might be possible via a `prebuild` statement in the build recipe. But I rather leave details to our packagers. Just make sure to point that out in your RFP then (or if you know how to do that, include it with your build recipe in the MR).

**Comment by mirfatif** [comment]
https://gitlab.com/fdroid/rfp/-/issues/1553

**Comment by mirfatif** [comment]
@IzzySoft thanks a lot for your generous help, guidance and efforts.

**Comment by IzzySoft** [comment]
As always: gladly done! :hugs:

**Comment by aitemoure1979** [comment]
ايد بعد النقود ليس لدي مال هل من مساعدة

في الاثنين، 29 مارس، 2021 01:08 Bouddi Mohamed <
***@***.***> كتب:

**Comment by aitemoure1979** [comment]
Smash l'età colche peratet

في الخميس، 25 مارس، 2021 20:07 Izzy ***@***.***> كتب:

**Comment by IzzySoft** [comment]
PS: Now that it's listed on "F-Droid proper", do you still want me to keep it in my repo for some reason? Usually I remove apps at this stage (see [here for some background](https://gitlab.com/IzzyOnDroid/repo/-/wikis/FAQ)).

**Comment by mirfatif** [comment]
> PS: Now that it's listed on "F-Droid proper", do you still want me to keep it in my repo for some reason? Usually I remove apps at this stage (see [here for some background](https://gitlab.com/IzzyOnDroid/repo/-/wikis/FAQ)).

It's totally up to you. If it's beneficial in anyway for the audience, you or me, you may consider keeping it. I'm deeply grateful, whatever you opt :)

**Comment by IzzySoft** [comment]
Eh, you must "demand" it – default is "remove". The link gives you some arguments, pro and con. PermissionManagerX is comparatively small (2.6M) so I wouldn't mind keeping it if you wish (with my repo it's mostly the size that matters in this context). So: Faster updates, easier migration for users fleeing the G-spot (oops) – at the price that folks *accidentally* install from my repo instead of from F-Droid? Then I'd say "keep" :see_no_evil:

**Comment by mirfatif** [comment]
I leave it to you because I'm quite unclear if keeping apps in your repo serves any of your interest or if it's an unnecessary burden on your resources. I don't have any stats either about how many downloads it's getting from your repo or from F-Droid's. If you've no problem with it then definitely keep it. Why won't I want it to be on your repo.

> at the price that folks _accidentally_ install from my repo instead of from F-Droid

That'd happen only if the user (trusts you and) has intentionally added your repo to F-Droid client. So it's not an accident, it's a choice. User is paying that price deliberately.

Btw if your repo is added to F-Droid client it shows the PMX app only from your repo, not from F-Droid repo. What's the logic to handle conflict of same package name from different repos?

**Comment by IzzySoft** [comment]
> burden on your resources

It starts becoming a burden if a single APK exceeds about 10M. Yours is far from that.

> any stats either about how many downloads it's getting from your repo

* 14 downloads between 2021-04-06 00:00 and 2021-04-07 20:00 CEST
* 40 downloads between 2021-03-29 00:00 and 2021-04-06 00:00 CEST

That's as far back as I can check my logs. I cannot tell for F-Droid.org as I cannot access the logs there.

> If you've no problem with it then definitely keep it.

There it is – then "keep" it is. I wouldn't ask if that were no option. Setting it to "keep" then, with the proper hint being shown.

> if your repo is added to F-Droid client it shows the PMX app only from your repo, not from F-Droid repo

Sure? I see both in my client. Open the entry, scroll down to "Versions", and tap the triangles to reveal details. Currently, there should be 2 entries for v1.05 – one from F-Droid, one from my repo.

> What's the logic to handle conflict of same package name from different repos?

Signature check? If you installed the APK with your signature, F-Droid's signature would count as "incompatible" – so if you have "show incompatible apps" disabled (which it is by default), that would explain why the F-Droid version is not shown. I don't have the app installed on the device I just checked with, so both are shown.

**Comment by mirfatif** [comment]
> Sure? I see both in my client.

Sorry about that. There seems something broken, on my device may be. Disabled/enabled both repos and now it's showing only F-Droid's version. May be repos aren't updating properly. I'll check and we may discuss later if I find something abnormal.

**Comment by IzzySoft** [comment]
No prob. As pointed out, my guess is you have the app installed from either Play, Github or my repo (i.e. the APK ***signed with your key***) – and, as per default, "show incompatible apps" dis… err, it shows only F-Droid's version? That would suggest the other way 'round. But it shouldn't switch between the two :thinking: Strange, indeed…

**Comment by aitemoure1979** [comment]
Ok

**Comment by mirfatif** [comment]
@IzzySoft sorry to bother you.

F-Droid seems to be doing something strange with app descriptions. See [this one](https://www.f-droid.org/en/packages/superfreeze.tool.android/) for instance.

I've made some changes to `full_description.txt`. But I think it's gonna look weird, mainly because of every linebreak being converted to a `br` tag (as per [documentation](https://f-droid.org/en/docs/All_About_Descriptions_Graphics_and_Screenshots/#html-descriptions)).

Is there a way to test the formatting other than waiting for the next release on F-Droid?

What format is supported on [your repo](https://apt.izzysoft.de/fdroid/index/apk/com.mirfatif.permissionmanagerx)? HTML, markdown, or both?

Another thing,  how to add website and translation links to build metadata?

Thanks

**Comment by IzzySoft** [comment]
> Is there a way to test the formatting other than waiting for the next release on F-Droid?

Not that I'm aware of. Of course you could try running `nl2br` on the file and view the result in a web browser, but that's not exactly the same.

> What format is supported on your repo? HTML, markdown, or both? 

HTML, Markdown, Plaintext (`nl2br`). Other than F-Droid, I configure that per-app so I can match what was set up with Fastlane. I usually run it through Markdown (which permits HTML), and add some sanitizing. The result usually looks much better than F-Droid's :wink: I've collected some formatting hints [in my wiki](https://gitlab.com/IzzyOnDroid/repo/-/wikis/Fastlane).

> I noticed that a line break is added after Features in F-Droid description but not in your repo. While asterisks are converted to bullet points in your repo, but not on F-Droid.

F-Droid only applies `nl2br`. I render Markdown (but store the results as HTML locally, so the index picks them up correctly). A single line break does not result in a new line; if you want a new line, that needs a separating empty line, per Markdown specs.

So, now it's time for me to ask: Why does F-Droid have a whatsNew but my repo has not? That's the draw-back with my setup: as I adjust to the app's offered "Fastlane feature set", I have to set all things manually. You've added `changelogs/` at some point, so my config didn't reflect that. Updated now. Need to check with my "Fastlane checker" why it didn't report that to me.

**Comment by IzzySoft** [comment]
> the icon at start makes the indentation a bit asymmetrical

Yepp. Work around is prepending a paragraph (and not starting with a list right away). Should look fine in the client, though.

> Did you do manual adjustments?

On each update? :rofl: I just run a Markdown-to-HTML converter on the text, with some sanitizing (so nobody injects e.g. Javascript). And prepend the icon 48x48.

> Is there 501 bytes limit?

No, but as [my Fastlane Cheat Sheet](https://gitlab.com/snippets/1895688) will inform you, there's a 500 char limit.

**Comment by mirfatif** [comment]
@IzzySoft It's malformed as expected: https://f-droid.org/packages/com.mirfatif.permissionmanagerx/

I'm going with `<p>` tags, and no new line in the [whole file](https://github.com/mirfatif/PermissionManagerX/blob/master/fastlane/metadata/android/en-US/full_description.txt).

I want to add some fields (`AuthorEmail`, `Donate` etc.) to [build metadata](https://gitlab.com/fdroid/fdroiddata/-/blob/master/metadata/com.mirfatif.permissionmanagerx.yml). Should I send PR?

Thanks for your availability :)

**Comment by IzzySoft** [comment]
> I'm going with `<p>` tags, and no new line in the whole file.

LOL – hard to read, but that should do it, yes. And yes, you can send the MR (if it's for F-Droid) or let me know what shall be put and I'll do it (as I have to do it anyway for my repo I guess).

**Comment by mirfatif** [comment]
> or let me know what shall be put and I'll do it

Thank you. Here it is: https://gist.githubusercontent.com/mirfatif/2f09509efba111ecc5ba3e5ec4b8961e/raw

**Comment by IzzySoft** [comment]
And [here you go](https://gitlab.com/fdroid/fdroiddata/-/merge_requests/9871).

As for version info: no API for that, but you could grab the JSON index and parse that. How often would you call that for parsing? Because downloading the entire index on each page view might be some overkill :wink:

Alternative: fetch `https://apt.izzysoft.de/fdroid/index/apk/com.looker.droidify?lang=en` and use RegEx: `!<tr><td><b>LastVersion:</b></td><td>([^<]+)</td></tr>!i` – though that would only be reliable as long as I don't change the page design (no plans for that, so it might be good enough).

**Comment by mirfatif** [comment]
Aah! My bad. How silly I'm. `Donate` and `Bitcoin` were not meant to be put in NUH. I want it to be free in true sense. Copy pasting is bad. Please remove it if you can. I hope you won't scold me :cry:

I was thinking about using this [JSON endpoint](https://shields.io/endpoint) to create a button like F-Droid offers. But it seems complicated without exposing a specific API for this purpose.

**Comment by IzzySoft** [comment]
https://gitlab.com/fdroid/fdroiddata/-/merge_requests/9873 – MR number 4. Thanks for helping me getting the Hacktoberfest swag :yum:

And sorry, but I have no endpoint to provide, and neither does F-Droid (to my knowledge).

**Comment by mirfatif** [comment]
> Thanks for helping me getting the Hacktoberfest swag yum

Forced education :)  
Thanks.

> neither does F-Droid (to my knowledge)

Talking about this: https://img.shields.io/f-droid/v/com.mirfatif.permissionmanagerx.svg

**Comment by IzzySoft** [comment]
Oh, didn't know about that. OK, digging in a little, they originally did exactly what I proposed: parsing the details page of the app using RegEx – see [here](https://github.com/niccokunzmann/shields/commit/0a040e9248fb6ea71888d288d3e12420fda2b699) for that approach. But [the final implementation](https://github.com/badges/shields/tree/master/services/f-droid) is using some API at F-Droid which I didn't know exists. Looking at [an example result](https://f-droid.org/api/v1/packages/org.fdroid.fdroid), this JSON isn't too hard to construct out of an existing JSON index.

Be welcome opening an issue with a feature request [with my repo's repo](https://gitlab.com/IzzyOnDroid/repo) – I cannot promise anything, but when I find time I could see if I can implement that for my repo, too. I assume I could even simply take the code from F-Droid (same backend, same index structure) if I knew where it is located – but am not sure if that could run on my server directly: F-Droid mostly uses Python, my server rather PHP…

**Comment by mirfatif** [comment]
@IzzySoft sorry to bother you once again. It's not gonna stop unless you stop responding :)

Upcoming release of PMX shows a button (highlighted in the screenshot) on long pressing an app which opens a third party app which is not open source. If the app is not installed on the device, a SnackBar is shown with an install button.

Does this fall under the category of [Non-Free Addons](https://f-droid.org/docs/Anti-Features/#NonFreeAdd) or any other anti-feature?

You can refer me to any detailed documentation if available, or a platform where this should be discussed.

Thanks.

<img src="https://user-images.githubusercontent.com/33040459/143720366-05e8db3a-257d-4707-93a6-51ccf9d79265.png" width="300"/>

**Comment by IzzySoft** [comment]
Yes, that's the way it is handled (see e.g. [here in the documentation](https://f-droid.org/docs/Anti-Features/#NonFreeAdd). But let's take a closer look at the phrasing. Back when I initialized my descriptions from what F-Droid had, it was

> The application promotes non-free add-ons, such that the app is effectively an advert for other non-free software and such software is not clearly labelled as such.

At F-Droid, this was meanwhile rephrased to:

> This Anti-Feature is applied to apps that, although Free Software themselves, promote other Non-Free applications or plugins.

So the definition has been widened (the restriction of "such that the app is effectively an advert" has been dropped).

Now the question is: Do you *promote* that non-free app? Is it such a central part? Where is it installed from? A clear way out (at least in my humble opinion) is to show that button only if the app was found on the device; in that case your app does not *promote* it but uses what's present. Point out the fact in your Readme, so it's your Readme that "promotes" (if one wants to call it that) but not your app. Or maybe there's a free app that could substitute?

For discussion, there's always the forum…

**Comment by mirfatif** [comment]
> Do you promote that non-free app?

Showing the "Install" button, I can't say I'm not promoting it.

> Is it such a central part?

Not at all. But it's closely related. That's how I use it personally. So want to make it public for others' convenience. Users will be able to see the native (Linux) process of an app to find out if it's running or not, and if running, in what state it is, how much resources it's using etc.

> Where is it installed from?

It's my own app named WhatsRunning, to be released on Play Store (a free and a paid/pro version), if not gets rejected ;)
A ReadMe and basic documentation as well as reporting issues (and possibly releases too) will be available on Github.

> A clear way out (at least in my humble opinion) is to show that button only if the app was found on the device

Yeah. That's the safer approach.

> Point out the fact in your Readme

I'll.

> Or maybe there's a free app that could substitute?

I'm not aware of any such app. You are welcome to test and give your opinion as well :)

**Comment by IzzySoft** [comment]
If it's your own app, open-sourcing it would clearly solve the dilemma :stuck_out_tongue: Until then, the "safer approach" is fully acceptable IMHO.
